book/bundles.rst

```diff
@@ -0,0 +1,175 @@
+.. index::
+   single: Bundles
+
+.. _page-creation-bundles:
+
+The Bundle System
+-----------------
+
+A bundle is similar to a plugin in other software, but even better. The key
+difference is that *everything* is a bundle in Symfony, including both the
+core framework functionality and the code written for your application.
+Bundles are first-class citizens in Symfony. This gives you the flexibility
+to use pre-built features packaged in `third-party bundles`_ or to distribute
+your own bundles. It makes it easy to pick and choose which features to enable
+in your application and to optimize them the way you want.
+
+.. note::
+
+   While you'll learn the basics here, an entire cookbook entry is devoted
+   to the organization and best practices of :doc:`bundles </cookbook/bundles/best_practices>`.
+
+A bundle is simply a structured set of files within a directory that implement
+a single feature. You might create a BlogBundle, a ForumBundle or
+a bundle for user management (many of these exist already as open source
+bundles). Each directory contains everything related to that feature, including
+PHP files, templates, stylesheets, JavaScripts, tests and anything else.
+Every aspect of a feature exists in a bundle and every feature lives in a
+bundle.
+
+An application is made up of bundles as defined in the ``registerBundles()``
```

Comment:
Instead of saying "an app is made of bundles defined in registerBundles()`, what about explaining it in reverse: "Bundles used in your applications must be enabled by registering them in the registerBundles() method."